pkg/volume/nfs/nfs.go

```diff
@@ -237,24 +237,24 @@ func (c *nfsUnmounter) TearDown() error {
 func (c *nfsUnmounter) TearDownAt(dir string) error {
 	notMnt, err := c.mounter.IsLikelyNotMountPoint(dir)
 	if err != nil {
-		glog.Errorf("Error checking IsLikelyNotMountPoint: %v", err)
+		glog.Errorf("nfs: Error checking IsLikelyNotMountPoint: %v", err)
 		return err
 	}
 	if notMnt {
-		return os.Remove(dir)
+		return os.RemoveAll(dir)
 	}
 
 	if err := c.mounter.Unmount(dir); err != nil {
-		glog.Errorf("Unmounting failed: %v", err)
+		glog.Errorf("nfs: Unmounting failed: %v", err)
 		return err
 	}
 	notMnt, mntErr := c.mounter.IsLikelyNotMountPoint(dir)
 	if mntErr != nil {
-		glog.Errorf("IsLikelyNotMountPoint check failed: %v", mntErr)
+		glog.Errorf("nfs: IsLikelyNotMountPoint check failed: %v", mntErr)
 		return mntErr
 	}
 	if notMnt {
-		if err := os.Remove(dir); err != nil {
+		if err := os.RemoveAll(dir); err != nil {
```

Comment:
@saad-ali Thank you for your review. I made the couple of changes to nfs.go to mirror the glusterfs plugin where, 1) the plugin name, "nfs" appears in log messages, and 2) os.RemoveAll is called instead of just os.Remove. I agree with your comment that the dir should be empty and that it is more destructive to RemoveAll. Would not the same argument apply to all shareable volumes, eg. apply to gluster too? I have no problem leaving in os.Remove.